Tests/PenguinGraphTests/GraphTransformationsTests.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import PenguinGraphs
+import XCTest
+
+final class GraphTransformationsTests: XCTestCase {
+
+  func testUnionEdges() {
+    var g1 = BidirectionalAdjacencyList<Int, String, Int>()
+    _ = g1.addVertex(storing: 10)
+    _ = g1.addVertex(storing: 11)
+    _ = g1.addVertex(storing: 12)
+    _ = g1.addEdge(from: 0, to: 1, storing: "0->1 (g1)")
+
+    var g2 = BidirectionalAdjacencyList<Empty, String, Int>()
+    _ = g2.addVertex()
+    _ = g2.addVertex()
+    _ = g2.addVertex()
+    _ = g2.addEdge(from: 1, to: 2, storing: "1->2 (g2)")
+
+    var g = g1.unionEdges(with: g2)
+
+    XCTAssertEqual(3, g.vertexCount)
+    XCTAssertEqual(Array(0..<3), Array(g.vertices))
+    XCTAssertEqual(Array(10..<13), g.vertices.map { g[vertex: $0] })
+
+    var recorder = TablePredecessorRecorder(for: g)
```

Comment:
How is this part testing `unionEdges`?  Just exercising code is OK, I guess, but it's really not clear what you're trying to discover by doing a BFS.  What you should do is build some generalized semantic tests for the protocols you claim to be modeling, like StdlibProtocolSemanticsTests.swift does, and use those, and test any APIs on the type (such as construction) that are not related to a protocol, and separately run BFS through some tests on a broad enough range of graph structures that you're confident it's handling genericity correctly.  I don't think using BFS on this particular data type (especially when it's only got 2 edges) is adding much.  WDYT?